### PR TITLE
fix(profile): handle missing GPU detection, expand GPU table, and scale throughput

### LINKED ISSUE #830: soup profile says "Fits in 24 GB VRAM" on a machine with no GPU, rejects --gpu rtx3050/rtx5070/rtx5080/b200, and uses A100 throughput for every card
## The defect

`soup profile` has three separate problems:

1. When no GPU is detected it silently assumes 24 GB and reports the fit as OK.

2. Its GPU table has no RTX 3050, no 50-series card below the 5090, no Blackwell datacenter parts and no laptop variants.

3. Its throughput estimate is an A100 lookup that does not depend on the GPU at all.

## Evidence

- `commands/profile.py:126-127`, the fallback when auto-detection finds no device:

  ```python

  # Default to 24 GB (common consumer GPU)

  return 24.0

  ```

  Nothing in the output says the number was assumed.

- `utils/profiler.py:6-31` (`GPU_MEMORY`) has `rtx5090: 32` and none of `rtx3050`, `rtx5060`/`5070`/`5080`, `b200`. Every key is a desktop model, and laptop parts differ: the RTX 5070 Laptop has 8 GB where the desktop RTX 5070 has 12 GB, and the RTX 4090 Laptop has 16 GB where the desktop has 24 GB.

- `estimate_speed` (`utils/profiler.py:222-255`) takes model size, quantization and batch size only. Its docstring says "Based on typical A100 throughput". `--gpu` changes the memory verdict but never the speed or time estimate.

## Reproduction

`examples/configs/sft_basic.yaml` (TinyLlama-1.1B, `quantization: none`) on an RTX 5070 Laptop (8151 MiB, driver 616.92). Panel padding trimmed.

With CUDA hidden, which is the same state as a CPU-only torch (PyPI's default on Windows):

```

$ CUDA_VISIBLE_DEVICES=-1 soup profile --config examples/configs/sft_basic.yaml

│ Tokens/sec: ~2,500                                                    │

│ OK Fits in 24 GB VRAM                                                 │

│ OK Recommended batch_size: 8                                          │

```

With the card visible:

```

$ soup profile --config examples/configs/sft_basic.yaml

│ Tokens/sec: ~2,500                                                    │

│ OK Fits in 8 GB VRAM                                                  │

│ OK Recommended batch_size: 2                                          │

```

Same throughput in both cases, and a batch-size recommendation 4x profile's own answer for this card.

```

$ soup profile --config examples/configs/sft_basic.yaml --gpu rtx5070

Unknown GPU: rtx5070

Valid options: a10, a100, a100_40gb, a30, a40, h100, h200, l4, l40, l40s, rtx3060, rtx3070, ...

```

The same `Unknown GPU` (exit 1) comes back for `--gpu rtx3050`, `rtx5080`, `b200` and `"rtx 5070 laptop"`.

## Why it matters

`soup profile` is the documented "before training" check. An OK that rests on assumed hardware is worse than no estimate: the failure just moves to an OOM at step 0. The time estimate is printed as a single number, so someone budgeting a run on a T4 is shown A100 wall-clock. The cards that are missing (3050, 5060-5080, laptop parts) are exactly the ones where a memory estimate decides whether a run is possible at all.

## Proposed fix

- When no device is detected and `--gpu` is not given, say so. Either withhold the fit verdict or label it "assuming 24 GB (no GPU detected; pass --gpu)".

- Extend `GPU_MEMORY` with `rtx3050` (plus the 6 GB variant), `rtx5060`/`5060ti`/`5070`/`5070ti`/`5080`, `b200`/`gb200`, and explicit laptop keys where memory differs. Accept the device name as torch reports it ("NVIDIA GeForce RTX 5070 Laptop GPU").

- Scale `estimate_speed` by a per-GPU relative throughput, or use `layer_stream_runtime.measure_gemm_tflops` when a GPU is present, and print a bracket rather than a point.

## Acceptance criteria

- [ ] A test that makes `get_gpu_info` report no device, runs `soup profile` without `--gpu`, and asserts the output does not present "Fits in 24 GB" as a measured verdict. It fails on current code.

- [ ] `--gpu rtx3050`, `--gpu rtx5070`, `--gpu "rtx 5070 laptop"` and `--gpu b200` are accepted with the right memory.

- [ ] `estimate_speed` returns different values for `t4` and `h100` on the same config.

---

**Related, filed the same day:**

- #829 — soup doctor shows only GPU name and memory: no compute capability, no check that the torch build supports t...
- #831 — run_cost prices an RTX A4000 as an A40 and an A100-SXM4-80GB as an A100 40GB; no rate for L4, H200, B200 or...

### PR DESCRIPTION
**Pull request by Guts1005**
Closes #830

### Summary

Resolves three issues in `soup profile`:
1. **Unmeasured fallback provenance**: When auto-detection fails (CPU-only PyTorch or no CUDA devices) and `--gpu` is omitted, the profiler explicitly labels the estimate as `Assuming 24 GB (no GPU detected; pass --gpu)` instead of presenting it as a measured `OK Fits in 24 GB VRAM`.
2. **GPU table & device string normalization**: Added lookup support for RTX 3050 (8GB/6GB), RTX 50-series (5060, 5060 Ti, 5070, 5070 Ti, 5080), Blackwell (B200, GB200), and mobile/laptop variants where VRAM differs from desktop parts. Added `normalize_gpu_name` to handle raw PyTorch device strings (e.g. `NVIDIA GeForce RTX 5070 Laptop GPU`) and hyphenated/spaced user flags.
3. **GPU-aware throughput scaling**: `estimate_speed` now scales estimated tokens/sec based on relative GPU throughput multipliers rather than returning fixed A100 throughput regardless of target card.

### Test Coverage

Added `TestIssue830GpuProfileDefects` in `tests/test_profile.py`:
- `test_issue830_gpu_lookup_and_variants`: Asserts memory lookup for RTX 3050, 5070, laptop variants, raw torch device strings, and B200.
- `test_issue830_estimate_speed_gpu_scaling`: Asserts throughput scaling differs across hardware tiers (e.g. T4 < A100 < H100).
- `test_issue830_no_gpu_detected_profile_warning`: Mocks no GPU detected and asserts the assumption warning is printed without presenting a false measured fit.

### CODE REVIEW

**Comment by Guts1005** [comment]
Closing in favor of #920 which was already in flight and actively in review with the maintainer. Thanks!
